rust/core-lib/src/tabs.rs

```diff
@@ -462,6 +479,69 @@ impl<W: Write + Send + 'static> Documents<W> {
     }
 }
 
+// =============== Fuchsia-specific synchronization plumbing
```

Comment:
I like breaking this stuff out (including the `use`s) but maybe go a step further and put all of the fuschia specific stuff at the bottom (before tests) of the file, where possible? Although I get keeping related impls together too.